### DISCUSSION TITLE
Aliases to permit use of more readable & native macOS key modifier names

### DISCUSSION THREAD

**Discussion by brandonzylstra** [zed-industries/zed#36944] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Zed's key mapping ability is powerful and flexible, but there is one simple thing--which I expect is very easy to implement--which it lacks.

In the keymap.json file for Zed several key modifiers are named with names that are foreign to macOS.  There are already aliases to accommodate Windows and Linux users, so that they don't have to do as much mental translation between the key names they are familiar with and the key names used by Zed, but no aliases to accommodate Mac users.

Apple keyboards in English have the following modifier keys that can be combined with regular keys when defining keystrokes:

- control
- command
- option
- shift

Unfortunately when I configure Zed I must use non-native names:

- alt instead of option
- ctrl instead of control
- cmd instead of command

I understand that there exist some foreign-language keyboards that use "ctrl", "cmd", but to my knowledge there are no English language keyboards from Apple that use these abbreviations.  Certainly every Apple keyboard that I have ever seen spells out "control" and "command".  

There are also some Apple keyboards that include "alt" in small letters on the "option" key, as an accommodation for switchers or users who install Windows or Linux as a secondary OS on their Apple hardware.  But "alt" is secondary and on recent keyboards that I own it is no longer used.  Every Apple keyboard has an "option" key, as that is the name the key has had on Apple keyboards going back to before Bootcamp existed, and before Apple was trying to appeal to switchers.

To give a concrete example, this currently works inside my keymap.json file:

```json
"ctrl-alt-v": "workspace::ToggleVimMode"
```

but if I try to use what is much more natural for me--both to write and to read--it fails:

```json
"control-option-v": "workspace::ToggleVimMode"
```

I must remove the names that are actually used on my keyboard and used in the English language and instead replace them with arbitrary alternatives that greatly slow down the speed at which I can review mappings in my keymap.json file to think about what I want to keep and what I want to change.

---

For reasons I can only guess, Apple has chosen to use abbreviations for the function key and the escape key.  In the interest of readability--[in the same way that Zed's keybindings documentation spells out the word "Escape"](https://zed.dev/docs/key-bindings), it would be helpful if there were also aliases for these keys, to make our keymap.json files as easy to read as possible.   However in this case it is definitely worth keeping `fn` and `esc` since the spelling used on the keyboard itself would ideally be available along with the more human-readable aliases.

**Comment by debuxxed** [comment]
it doesn't work the other way: i have the keymap created from MacOS, but it doesn't work on Linux for me.

E.g. if i press "Ctrl+L" it doesn't translate to "Cmd+L" i have in keymap file
